### PR TITLE
Geyser rebalancing, gives miners points for discovering geysers

### PR DESCRIPTION
**Pull request by Time-Green**
Miners can now hit geysers with their mining scanner to give away the geyser type, give them some mining points based on rarity and mark it on the GPS system.

Removes the ClF3 geyser, adds the strange geyser (yes it's just a random reagent, maintpill/strange seed style).

Removes the reinforced plunger, you can now activate geysers with a normal plunger

**Point rewards for the geysers:**
Plasma-oxide: 100
Protozine: 100
Hollow water: 100
Wittel: 250
Strange: 500cr

**Geyser weight:**
Protozine: 10
Plasma-oxide: 10
Hollow water: 10
Wittel: 6
Strange: 2

:cl: Time-Green
add: Miners can now discover geysers with their mining scanner for a point reward! This will discover their type and mark that geyser (with type) on the GPS system.
del: removes the ClF3 geysers
add: Adds a very rare strange geyser that has a random reagent (maintpill/strange seed style)
balance: Geysers can now be activated with normal, cheap plungers
/:cl:

### Why
Geysers are a fucking joke. You need to pay a ridiculous amount of money, search lavaland for uncommon geysers, with absolutely nothing to make the ride worthwile. Wittel was the most interesting and rare, while the rest was just a waste of your round. Metalgen was nerfed to need a secret recipy, making geysers even less reliable.

Geysers are now not locked behind a massive money paywall, which was designed with paychecks in mind. 

Miners can now assist chemists by marking geysers and get incentivized to do so, making a chemists decision to work with geysers a lot more reliable since you already know what you're gonna get in most cases. 

Buffing the availability of wittel makes it so people can actually create the recipys it's designed with. Penthrite, noartrium and gravitum are all really fun and unique chems, but they might aswell not even exist because wittel is practically completely unattainable. This still makes them require dedication to make, but makes it actually something a chemist could feasibly do. They get warned by a miner, decide wheter to go for it or not, take a GPS, setup a plumbing teleporter and make some unique and rare chems on station. 

(I also really wanted to make geysers spit acid when people got near them, but I'd need to make a component/element that would listen to a range of tiles ON_STEP signal and properly handle everything that comes along with doing that, and I just... meh)

### CODE REVIEW

**Comment by itseasytosee** [comment]
A mining point reward would probably make more sense than a cash reward.

**Comment by tralezab** [comment]
Ogey

**Comment by Nexofas** [comment]
Rather than credits mining points would be more fitting. Its a good idea.

**Comment by Time-Green** [comment]
Fuck I thought mining points and credits were merged, ugh...

**Comment by Time-Green** [comment]
Yeah okay it's mining points now

**Comment by Nexofas** [comment]
> Fuck I thought mining points and credits were merged, ugh...

You can pay 2000 mining points for 1000 credits. Doubling those values in order to exchange them as mining points may work.

**Comment by ATH1909** [comment]
thank christ

**Comment by D4C-420** [comment]
>  do chemists have easy access to gpses?

There is a few in the public area on lava land so yes.

**Comment by Time-Green** [comment]
> do the miners have to manually notify the chemists when a new geyser is analyzed? 

Yeah

**Comment by moocowswag** [comment]
> > do chemists have easy access to gpses?
> 
> There is a few in the public area on lava land so yes.

Chemists will probably struggle to even get to lavaland on tramstation since there is (apparently) no public mining shuttle.

File: code/game/objects/structures/lavaland/geyser.dm [hunk 1]
```diff
@@ -46,14 +64,57 @@
...
+
+			AddComponent(/datum/component/gps, true_name) //put it on the gps so miners can mark it and chemists can profit off of it
+
+			if(isliving(user))
+				var/mob/living/living = user
+
+				var/obj/item/card/id/card = living.get_idcard()
+				if(card)
+					to_chat(user, "<span class='notice'>[point_value] mining points have been paid out!</span>")
+					card.mining_points = point_value
```
Comment (GoldenKeyboard) [inline code/game/objects/structures/lavaland/geyser.dm:88]:
```suggestion
					card.mining_points += point_value
```

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/lavaland/geyser.dm [hunk 2]
```diff
@@ -46,14 +64,57 @@
 	if(do_after(user, 50 * P.plunge_mod, target = src) && !activated)
 		start_chemming()
 
+/obj/structure/geyser/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/mining_scanner) || istype(I, /obj/item/t_scanner/adv_mining_scanner))
```
Comment (Mothblocks) [inline code/game/objects/structures/lavaland/geyser.dm:68]: early return

File: code/game/objects/structures/lavaland/geyser.dm [hunk 3]
```diff
@@ -46,14 +64,57 @@
...
+		if(discovered)
+			to_chat(user, "<span class='warning'>This geyser has already been discovered!</span>")
```
Comment (Mothblocks) [inline code/game/objects/structures/lavaland/geyser.dm:70]: early return

**Review by Mothblocks — CHANGES_REQUESTED** [review]
nooartrium is wicked overpowered and i suspect this pr will be the first step in its demise but dont say i didnt tell you so

**Comment by Time-Green** [comment]
> nooartrium is wicked overpowered and i suspect this pr will be the first step in its demise but dont say i didnt tell you so

Probably. It's basically not existed before, except for maybe maintpill / strange seeds. I'd prefer to see how it's used and all. Most antags wont do it because it's too much work, and non-antags generally don't have any purpose for it except to meme and maybe use it as a special strange reagent

**Comment by Mothblocks** [comment]
nooartrium is the one that makes it so you cant go into hard crit or something

**Comment by Time-Green** [comment]
> nooartrium is the one that makes it so you cant go into hard crit or something

It also revives you and makes it so you can’t die until you either hit 500 damage or your heart gives out. I’ve never seen it in action, but it’s one of those chems that requires a level of stability in the round that you probably wont be able to use it when you actually need it, except with maybe fucking around and turning the clown into a walking zombie

We’ll see soon enough how it’s used I guess

**Pull request merged**